doc/acl/Extend-L3V6ACLs.md

```diff
@@ -0,0 +1,431 @@
+New fields are added to the ACL capability in STATE_DB to help applications identify the platforms where the new L3V4V6 ACL table is supported; this is done for both ingress and egress individually.
+
+
+**Matches supported in L3V4V6 table**
+
+- VLAN_ID
+- *IP_TYPE**
+- *ETHER_TYPE**
+- SRC_IPV6
+- DST_IPV6
+- SRC_IP
+- DST_IP
+- ICMPV6_TYPE
+- ICMPV6_CODE
+- ICMP_TYPE
+- ICMP_CODE
+- IP_PROTOCOL
+- NEXT_HEADER
+- L4_SRC_PORT
+- L4_DST_PORT
+- TCP_FLAGS
+
+***Note**: It is recommended that every ACL Rule should include at least one of [*IP_TYPE*, *ETHER_TYPE*] in the matching criteria when matching of L3 header fields. When we handle unsupported platforms by adding two ACL tables in SAI, this allows orchestrator to decide in which underlying SAI ACL table to place the rule in. In phase-1, if these fields are not provided in the ACL rule, the underlying SAI platform must do one of the following: 
+1) install a single rule if the hardware supports it, or
+2) make multiple rules (one rule per IP type), or
+3) throw error as unsupported.
+
+**Actions allowed in the table of the type L3V4V6**
+
+The default actions supported in the new table type are the same as L3 and L3V6, viz.,
+
+- PACKET_ACTION 
+- REDIRECT_ACTION 
+
+Note: The above actions would be added to the ACL table based on the ACL capability. For example, on platforms where REDIRECT_ACTION is not supported in the egress direction, the only action supported will be PACKET_ACTION. This is similar to what is being done today for L3 and L3V6 ACL tables.
+
+**Bindpoints supported in the table of the type L3V4V6**
+The same as L3 and L3V6 ACL table, viz.,
+- Port
+- LAG
+
+##### Orchagent
+
+ The new ACL table type called L3V4V6 is created during Orchagent initialization.
+
+  ``` aclorch::init() -> initDefaultTableTypes() -> addAclTableType(TABLE_TYPE_L3V4V6) ```
+  
+Using platform specific checks, AclOrch identifies platforms where  L3V4V6 ACL tables can be supported. Creation of an ACL table of type L3V4V6 on an unsupported platform results in orchagent logging an error in the syslog. A SAI ACL table create call is done only on supported platforms; this avoids crashes on unsupported platforms. In the future, orchagent would internally create separate v4 and v6 ACL tables on these unsupported platforms.
+
+##### STATE_DB
+A new field called `supported_L3V4V6` is added to the ACL capability in STATE_DB. This field is set to true by orchagent during 'AclOrch init' on platforms where v4 fields and v6 fields can be matched in the same hardware ACL table. This provides an interface to the operator to identify the platforms where the new L3V4V6 ACL table is supported. In the future, this field will be used by Orchagent to implement a single user configured L3V4V6 ACL table as two underlying SAI ACL tables on platforms where v4 fields and v6 fields cannot be matched in the same hardware ACL table.
```

Comment:
what is the criteria for acl orch to identify the platforms that supports both v4 and v6 fields?

Reply:
@prsunny 

Unfortunately, the SAI community decided not to have such special ACL capability checks and instead continue with the model we are doing for Everflow/Mirror ACL tables.

[https://github.com/sonic-net/SONiC/pull/1267#discussion_r1143835883](https://github.com/sonic-net/SONiC/pull/1267#discussion_r1143835883)

The alternative of creating a new SAI API query for the SAI implementation to say supported or not for any given list of ACL match-fields will lead to a combinatorial explosion-- the SAI implementation needs to handle 2^n (where n is the number of ACL match fields supported) combinations.

We did not want to overload the existing platform checks used to detect the capability to combine v4 and v6 fields for Mirror ACL tables. So, orchagent is using platform checks to decide which platforms support combining v4 and v6 fields.